src/processor/java/org/truffleruby/processor/BuildInformationProcessor.java

```diff
@@ -56,10 +58,13 @@ public synchronized void init(ProcessingEnvironment env) {
         try {
             trufflerubyHome = findHome();
             buildName = System.getenv("TRUFFLERUBY_BUILD_NAME");
-            fullRevision = runCommand("git rev-parse HEAD");
+            fullRevision = runCommand("git rev-parse HEAD")
+                    .orElseThrow(() -> new Error("git rev-parse command failed"));
             shortRevision = fullRevision.substring(0, 8);
-            compileDate = runCommand("git log -1 --date=short --pretty=format:%cd");
-            copyrightYear = compileDate.split("\\-")[0];
+            isDirty = runCommand("git diff --quiet").isEmpty();
```

Comment:
Using the exit code seems more reliably than checking the output:
```
       --quiet
           Disable all output of the program. Implies --exit-code.
       --exit-code
           Make the program exit with codes similar to diff(1). That is, it exits with 1 if there were differences and 0 means no
           differences.
```
But it's probably fine as an approximation.

Reply:
I misread, it's Optional#isEmpty, all good :)